Sources/TuistAutomation/Device/PhysicalDevice.swift

```diff
@@ -3,10 +3,23 @@ import XcodeGraph
 
 /// Represents a physical device, such as an iPhone
 public struct PhysicalDevice: Codable, Equatable, Identifiable {
+    public enum TransportType: String, Codable {
+        case wifi
+        case usb
+        case unknown
+    }
+
+    public enum ConnectionState: String, Codable {
+        case connected
+        case disconnected
+    }
+
     public let id: String
     public let name: String
     public let platform: Platform
-    public let osVersion: String
+    public let osVersion: String?
+    public let transportType: TransportType
+    public let connectionState: ConnectionState
```

Comment:
When I tried this out, I got the `Data couldn't be read because it's missing` coming from [here](https://github.com/tuist/tuist/blob/main/app/TuistApp/Sources/Utilities/AppStorage.swift#L19-L22). I believe the issue is that we're saving the full `PhysicalDevice` to the user defaults and when the model is changed, the decoding breaks.

We should refactor the code to only read and write a `PhysicalDevice` ID instead. Let me know if you wanna tackle that, otherwise, I can take this one on in a separate PR.

Reply:
> We should refactor the code to only read and write a `PhysicalDevice` ID instead. Let me know if you wanna tackle that, otherwise, I can take this one on in a separate PR.

Sure, if you don’t have any related changes, I’d be happy to tackle this.

Reply:
I don't, that would be awesome!

You can reproduce this by:
- Running the app from `main`
- Selecting a device
- Running the app from this branch
- You should see the error as the decoding fails